No home, zoom, etc buttons on module window in Ubuntu

From this [forum post](http://forum.cellprofiler.org/t/zoom-pan-home-controls-in-the-modules-viewer-in-lunix/4552):

"I am glad that CellProfiler installation on Linux (Ubuntu 16.04) now works (Since version 2.2.0).

The CellProfiler behaves similar to the windows in most respects, however the modules windows lacks the controls associated with zooming, panning. These controls facilitate accurate pipeline parameters like thresh-holding, bounds, smoothing scales, etc.

Is that a known issue in Linux installation, or is there a problem in my installation/JRE/JVM?" 

![image](https://cloud.githubusercontent.com/assets/6721515/24662651/b31a84ce-1923-11e7-89ac-7cb7644ff06a.png)

Hi
I just updated the packages, using `--force-reinstall` flag, i.e. `pip install --upgrade --force-reinstall matplotlib`
Now the cell profiler is messed up, it does not show any GUI. the error now seems to be in Java. 
The entire error message is:
`
java.lang.IllegalArgumentException: Cannot handle replace call to list in ij.Menus's public static synchronized java.lang.String[] getPlugins()
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:725)
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:650)
	at net.imagej.patcher.LegacyExtensions.addExtraPlugins(LegacyExtensions.java:477)
	at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:175)
	at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
	at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
	at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
	at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
	at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
	at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 18
	at javassist.CtClassType.getClassFile2(CtClassType.java:203)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:110)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
	at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)
	at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
	at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
	at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
	at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.CodeGen.atIfStmnt(CodeGen.java:390)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:354)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.Javac.compileStmnt(Javac.java:568)
	at javassist.expr.MethodCall.replace(MethodCall.java:234)
	at net.imagej.patcher.CodeHacker$7.edit(CodeHacker.java:686)
	at javassist.expr.ExprEditor.loopBody(ExprEditor.java:191)
	at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
	at javassist.CtBehavior.instrument(CtBehavior.java:618)
	at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
	at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:670)
	... 17 more
Caused by: java.io.IOException: invalid constant type: 18
	at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
	at javassist.bytecode.ConstPool.read(ConstPool.java:970)
	at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
	at javassist.bytecode.ClassFile.read(ClassFile.java:693)
	at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
	at javassist.CtClassType.getClassFile2(CtClassType.java:190)
	... 37 more
java.lang.IllegalArgumentException: Cannot handle app name in ij.ImageJ's public <init>(java.applet.Applet applet, int mode)
	at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:446)
	at net.imagej.patcher.LegacyExtensions.insertAppNameHooks(LegacyExtensions.java:406)
	at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:291)
	at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
	at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
	at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
	at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
	at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
	at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
	at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 18
	at javassist.CtClassType.getClassFile2(CtClassType.java:203)
	at javassist.CtClassType.subtypeOf(CtClassType.java:303)
	at javassist.CtClassType.subtypeOf(CtClassType.java:318)
	at javassist.compiler.MemberResolver.compareSignature(MemberResolver.java:247)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:119)
	at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
	at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)
	at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
	at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
	at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
	at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
	at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
	at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
	at javassist.compiler.Javac.compileStmnt(Javac.java:568)
	at javassist.expr.MethodCall.replace(MethodCall.java:234)
	at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:427)
	at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:441)
	at javassist.expr.ExprEditor.loopBody(ExprEditor.java:219)
	at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
	at javassist.CtBehavior.instrument(CtBehavior.java:618)
	at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
	at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:402)
	... 16 more
Caused by: java.io.IOException: invalid constant type: 18
	at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
	at javassist.bytecode.ConstPool.read(ConstPool.java:970)
	at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
	at javassist.bytecode.ClassFile.read(ClassFile.java:693)
	at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
	at javassist.CtClassType.getClassFile2(CtClassType.java:190)
	... 37 more
Exception fetching new version information from http://cellprofiler.org/CPupdate.html: HTTP Error 404: Not Found
ilastik import: failed to import the ilastik. Please follow the instructions on
    "http://www.ilastik.org" to install ilastik
Traceback (most recent call last):
  File "/home/vs_lab_admin/CellProfiler/cellprofiler/modules/classifypixels.py", line 71, in <module>
    from ilastik.core.dataMgr import DataMgr, DataItemImage
ImportError: No module named ilastik.core.dataMgr
ilastik import: failed to import the ilastik. Please follow the instructions on
                          "http://www.ilastik.org" to install ilastik
Traceback (most recent call last):
  File "/home/vs_lab_admin/CellProfiler/cellprofiler/modules/ilastik_pixel_classification.py", line 70, in <module>
    import ilastik_main
ImportError: No module named ilastik_main
Failed to stop Ilastik
`

--- a/cellprofiler/gui/figure.py
+++ b/cellprofiler/gui/figure.py
@@ -524,6 +524,15 @@ class Figure(wx.Frame):
             return
         available_width, available_height = self.GetClientSize()
         nbheight = self.navtoolbar.GetSize()[1]
+
+        # On some operating systems (specifically Ubuntu), the matplotlib toolbar
+        # may not render with an appropriate height. This is the result of creating
+        # the navtoolbar without an appropriate slider. Until this is refactored
+        # to make it size properly, this check ensures the navtoolbar at least appears.
+        # https://github.com/CellProfiler/CellProfiler/issues/2679
+        if nbheight < 40:
+            nbheight = 40
+
         self.navtoolbar.SetPosition((0, 0))
         self.navtoolbar.SetSize((available_width, nbheight))
 

